integration/test/Test/LegalHold.hs

```diff
@@ -0,0 +1,86 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2023 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Test.LegalHold where
+
+import API.Brig
+import API.Common
+import API.Galley
+import API.GalleyInternal
+import Control.Lens ((.~), (^?!))
+import qualified Data.ProtoLens as Proto
+import Data.ProtoLens.Labels ()
+import GHC.Stack
+import Numeric.Lens (hex)
+import qualified Proto.Otr as Proto
+import qualified Proto.Otr_Fields as Proto
+import SetupHelpers
+import Testlib.MockIntegrationService
+import Testlib.Prelude
+
+testLHMessageExchange :: HasCallStack => App ()
+testLHMessageExchange = abstractTestLHMessageExchange True
+
+testLHMessageExchangeOldClients :: HasCallStack => App ()
+testLHMessageExchangeOldClients = abstractTestLHMessageExchange False
+
```

Comment:
```
    • No instance for (HasTests (Bool -> Testlib.Types.App ()))
```

hm, don't think so.  also, the more explicit way is nice because it allows me to only run one of them by name, and the names add explanation.

Reply:
I'm pretty sure that this works for toplevel tests...

Reply:
maybe not for `Bool` actually

Reply:
yeah sorry I misremembered